app/models/dataset/credential/CredentialService.scala

```diff
@@ -43,6 +43,8 @@ class CredentialService @Inject()(credentialDAO: CredentialDAO) {
           secret <- credentialSecret
           secretJson <- tryo(Json.parse(secret)).toOption
         } yield GoogleServiceAccountCredential(uri.toString, secretJson, userId.toString, organizationId.toString)
+      case _ =>
+        None
```

Comment:
this was where the match error originated in case of an unknown uri scheme. That shouldn’t happen.